### PR TITLE
Feature.resource docstrings 205

### LINKED ISSUE #205: Add documentation to resource.py

### PR DESCRIPTION
**Pull request by zancas**
@pjbreaux Please see the commit message for deets.

Fixes #205

### CODE REVIEW

**Comment by swormke** [comment]
Before you merge this I'd like to see what the generated docs look like. I don't think they will be what we had in mind for user facing docs.

**Comment by pjbreaux** [comment]
Ok, can you walk me through how to generate today?

**Comment by zancas** [comment]
@pjbreaux OK, now it's correct...  -er.   I can show you how to render the docstrings for review.

File: f5/bigip/resource.py [hunk 1]
```diff
@@ -14,28 +14,66 @@
...
-    * ResourceBase -- only `read` is generally supported in all resource types,
-      this class provides `read`. ResourceBase objects are usually instantiated
-      via setting lazy attributes. ResourceBase provides a contructor to match
-      the lazy constructor. The expected behavior is that all resource
-      subclasses depend on this constructor to correctly set their
-      self._meta_data['uri'].
+    * ResourceBase -- only `refresh` is generally supported in all resource
+      types, this class provides `refresh`. ResourceBase objects are usually
+      instantiated via setting lazy attributes. ResourceBase provides a
+      contructor to match its call in LazyAttributeMixin.__getattr__. The
```
Comment (pjbreaux) [inline f5/bigip/resource.py]: constructor

File: f5/bigip/resource.py [hunk 2]
```diff
@@ -14,28 +14,66 @@
...
 
+We use methods named Create, Refresh, Update, Load, and Delete to manipulate
+BigIP device services.
+
+Methods:
+
+  * create -- uses HTTP POST, creates a new resource and with its own URI on
+    the device
+  * refresh -- uses HTTP GET, obtains the state of a device resource, and sets
+    the representing Python Resource Object track device state via its attrs
```
Comment (pjbreaux) [inline f5/bigip/resource.py]: ...Object <to> track device state

**Comment by zancas** [comment]
@pjbreaux I addressed your typo fixes

**Comment by pjbreaux** [comment]
Add :inherited-members: for each module which inherits from resource.Resource so we can see the inherited attributes and methods in the documentation.

**Comment by pjbreaux** [comment]
The generated docs look good. I don't require any additional fixups.

**Pull request merged**